### ISSUE TITLE
Collection of enum's cannot be validated by Hibernate Validator

### ISSUE THREAD

**Issue by paulmiddelkoop** [joelittlejohn/jsonschema2pojo#518]
When generating a POJO with a collection of enums, Hibernate validator will log an exception during validation.

It's arguable if this is a Hibernate Validator or jsonschema2pojo problem. Since enum's cannot be validated in my opinion, it would be nice if jsonschema2pojo will not add a Valid annotation on collections of enums. 

Schema:
```javascript
{
  "$schema": "http://json-schema.org/draft-03/schema#",
  "type": "object",
  "javaType": "jsonschema.EnumArray",
  "properties": {
    "months": {
      "type": "array",
      "items": {
        "javaType": "java.time.Month"
      }
    }
  },
  "additionalProperties": false
}
```

Following program will log exception:
```java
package jsonschema;

import javax.validation.Validation;
import java.time.Month;
import java.util.Arrays;

public class ValidationError {

    public static void main(String[] args) {
        Validation.buildDefaultValidatorFactory().getValidator().validate(new EnumArray().withMonths(Arrays.asList(Month.JANUARY)));
    }
}

```
I'm using jsonschema2pojo-maven-plugin 0.4.21 with Hibernate Validator 5.2.4.Final.

Stacktrace:
10:14:05.066 [main] WARN  o.h.v.i.m.p.TypeAnnotationAwareMetaDataProvider - Constraints on the parameters of constructors of non-static inner classes are not supported if those parameters have a generic type due to JDK bug JDK-5087240.
java.lang.ArrayIndexOutOfBoundsException: 0
	at java.lang.reflect.Parameter.getAnnotatedType(Parameter.java:237)
	at org.hibernate.validator.internal.metadata.provider.TypeAnnotationAwareMetaDataProvider.findTypeAnnotationConstraintsForExecutableParameter(TypeAnnotationAwareMetaDataProvider.java:85)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.getParameterMetaData(AnnotationMetaDataProvider.java:495)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.findExecutableMetaData(AnnotationMetaDataProvider.java:344)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.getMetaData(AnnotationMetaDataProvider.java:329)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.getConstructorMetaData(AnnotationMetaDataProvider.java:307)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.retrieveBeanConfiguration(AnnotationMetaDataProvider.java:139)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.getBeanConfiguration(AnnotationMetaDataProvider.java:125)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.getBeanConfigurationForHierarchy(AnnotationMetaDataProvider.java:108)
	at org.hibernate.validator.internal.metadata.BeanMetaDataManager.createBeanMetaData(BeanMetaDataManager.java:166)
	at org.hibernate.validator.internal.metadata.BeanMetaDataManager.getOrCreateBeanMetaData(BeanMetaDataManager.java:194)
	at org.hibernate.validator.internal.metadata.BeanMetaDataManager.getBeanMetaData(BeanMetaDataManager.java:142)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraint(ValidatorImpl.java:783)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraints(ValidatorImpl.java:681)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:416)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validate(ValidatorImpl.java:205)
	at jsonschema.ValidationError.main(ValidationError.java:12)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)

**Comment by joelittlejohn** [comment]
Is this error thrown only when you include constructors in your generated output? I don't quite understand the error message here that refers to 'parameters of constructors'. It would be useful if you could show the Java that you believe to be causing the problem and explain how you'd like it to change.

**Comment by paulmiddelkoop** [comment]
As it turn out, this is actually a JDK bug which is already fixed when using JDK 1.8.0_40 or greater.
